src/renderer/redux/actions/search.js

```diff
@@ -5,63 +5,119 @@ import { doNavigate } from 'redux/actions/navigation';
 import { selectCurrentPage } from 'redux/selectors/navigation';
 import batchActions from 'util/batchActions';
 
-// eslint-disable-next-line import/prefer-default-export
-export function doSearch(rawQuery) {
-  return (dispatch, getState) => {
-    const state = getState();
-    const page = selectCurrentPage(state);
+// TODO: this should be in a util
+const handleResponse = response =>
+  response.status === 200
+    ? Promise.resolve(response.json())
+    : Promise.reject(new Error(response.statusText));
 
-    const query = rawQuery.replace(/^lbry:\/\//i, '');
+export const doSearch = rawQuery => (dispatch, getState) => {
+  const state = getState();
+  const page = selectCurrentPage(state);
 
-    if (!query) {
-      dispatch({
-        type: ACTIONS.SEARCH_CANCELLED,
-      });
-      return;
-    }
+  const query = rawQuery.replace(/^lbry:\/\//i, '');
 
+  if (!query) {
     dispatch({
-      type: ACTIONS.SEARCH_STARTED,
-      data: { query },
+      type: ACTIONS.SEARCH_CANCELLED,
     });
+    return;
+  }
 
-    if (page !== 'search') {
-      dispatch(doNavigate('search', { query }));
-    } else {
-      fetch(`https://lighthouse.lbry.io/search?s=${query}`)
-        .then(
-          response =>
-            response.status === 200
-              ? Promise.resolve(response.json())
-              : Promise.reject(new Error(response.statusText))
-        )
-        .then(data => {
-          const uris = [];
-          const actions = [];
+  dispatch({
+    type: ACTIONS.SEARCH_STARTED,
```

Comment:
Can you refactor these to the proper names while working on this?

Reply:
Yep